pkg/nvlib/device/device.go

```diff
@@ -208,6 +209,47 @@ func (d *device) IsMigEnabled() (bool, error) {
 	return (mode == nvml.DEVICE_MIG_ENABLE), nil
 }
 
+// IsFabricAttached checks if a device is attached to a GPU fabric.
+func (d *device) IsFabricAttached() (bool, error) {
+	if d.lib.hasSymbol("nvmlDeviceGetGpuFabricInfo") {
+		info, ret := d.GetGpuFabricInfo()
+		if ret == nvml.ERROR_NOT_SUPPORTED {
+			return false, nil
+		}
+		if ret != nvml.SUCCESS {
+			return false, fmt.Errorf("error getting GPU Fabric Info: %v", ret)
+		}
+		if info.State != nvml.GPU_FABRIC_STATE_COMPLETED {
+			return false, nil
+		}
+		if info.Status != uint32(nvml.SUCCESS) {
+			return false, fmt.Errorf("error getting GPU Fabric Info: %v", info.Status)
```

Comment:
We said no error here, just nil

Also, I'm surprised that info.Status is not of type `Return`. The Api documentation said it should be ...

Reply:
Yeah -- strange that this was converted to a uint32 ....
https://github.com/NVIDIA/go-nvml/blob/main/pkg/nvml/nvml.h#L2701

Reply:
yeah, it says is same type as return but looks like is not the case

Reply:
Given that I would cast it the other way around, i.e.
```
nvml.Return(info.Status) != nvml.SUCCESS
```

Reply:
```go
type GpuFabricInfo struct {
	ClusterUuid [16]uint8
	Status      uint32
	CliqueId    uint32
	State       uint8
	Pad_cgo_0   [3]byte
}
```

Reply:
thanks for the suggestion, edited